lib/lightspeed/request.rb

```diff
@@ -34,19 +34,32 @@ def initialize(client, method:, path:, params: nil, body: nil)
       @raw_request = request_class.new(uri)
       @raw_request.body = body if body
       @raw_request.set_form_data(@params) if @params && @method != :get
-      @raw_request["Authorization"] = "OAuth #{client.oauth_token}" if client.oauth_token
+      @client = client
+      set_authorization_header
```

Comment:
It's not a big deal, but personally I hate things that mutate variables as side effects. So I'd definitely much prefer keeping this on the same line.